### DISCUSSION TITLE
Route Handlers not the correct solution for form submissions?

### DISCUSSION THREAD

**Discussion by paulwongx** [vercel/next.js#48636] [Help]
### Summary

In the [documentation](https://beta.nextjs.org/docs/routing/route-handlers), it says the following about route handlers...

> Note: Previously, API Routes could have been used for use cases like handling form submissions. Route Handlers are likely not the solution for these uses cases. We will be recommending the use of [mutations](https://beta.nextjs.org/docs/data-fetching/mutating) for this when ready.

In light of the lack of a mutations feature currently, what is the drawback of using route handlers to submit form data via a POST route, or should one not use it for that at all? 

I'd like to get some clarity on the context for this statement - what is lacking if using it for this purpose, why is it not recommended, can you still use it though in the interim? 

My thinking is... because route handlers are in the app dir, meaning it wants to be server side rendered primarily.... but a form submission POST route fundamentally needs to be dynamically generated every time based on the input data (so can't be pre-rendered on the server) so it's the opposite of the what the appDir wants by default - hence the incongruency? So if this is the case, should we be using the `pages` folder to submit form data? Or does it not matter if we use route handlers and it's just not aligned principally , but very much still something you can do... As in... you can do whatever you want, it's just not recommended cuz there will be something better but in the future, but in the interim, it's fine...? But if we use it, we just need to use the `router.refresh()` pattern as suggested [here](https://beta.nextjs.org/docs/data-fetching/mutating) when POSTing?

**Comment by ghost** [comment]
I'm also trying to figure out why. Ultimately, I think it's because of you're trying to submit forms using them, you might end up creating `/api` folders in many different routes segments to contain the `route.js` file (since they can't coexist with `page.js`), as you would want to keep the logic for the form and its submission within the segment. Since they are now working on the solution to let you include the server side code for handling submissions in the `page.js` (server actions), they don't want to encourage the use of `route.js` files for that purpose, since it'll only lead to creating folders in your route segments that won't be pages. So they think since that is not even a part of the conventions, you might as well stick to what you would normally do to handle submissions if route handlers couldn't be used that way, while they work on the solution built exactly for that.

Just saying...
